feat(apple): Add docs for pre-warmed app starts (#5762)
getsentry/sentry-docs: src/platforms/apple/common/performance/instrumentation/automatic-instrumentation.mdx

### Prewarmed App Start Tracking

Starting with iOS 15, the system might [prewarm](https://developer.apple.com/documentation/uikit/app_and_environment/responding_to_the_launch_of_your_app/about_the_app_launch_sequence#3894431) your app by creating the process before the user opens it. In such cases, we won’t be able to reliably measure the app start. But with [sentry-cocoa 7.31.0](https://github.com/getsentry/sentry-cocoa/releases/tag/7.31.0), we’ve introduced a `enablePreWarmedAppStartTracking` feature, (still in its experimental phase), which allows us to collect prewarmed app starts.

Once enabled, the SDK will drop the first app start span if prewarming pauses. This approach will shorten the app start duration, but it will accurately represent the span of time from when a user clicks the app icon to when the app is responsive.
With this feature, the SDK differentiates between four different app start types:

* __Non-prewarmed cold start__ (See _cold start_  definition above.)
* __Non-prewarmed warm start__ (See _warm start_  definition above.)
* __Prewarmed cold start__

> * __Prewrmed warm start__

Reviewer: ```suggestion
* __Prewarmed warm start__
```

> You can filter for different app start types in Discover with `app_start_type:cold.prewarmed`, `app_start_type:warm.prewarmed`, `app_start_type:cold`, and `app_start_type:warm`.

Reviewer: Let's link the name of the feature in case someone isn't familiar with it yet
```suggestion
You can filter for different app start types in [Discover](/product/discover-queries/) with `app_start_type:cold.prewarmed`, `app_start_type:warm.prewarmed`, `app_start_type:cold`, and `app_start_type:warm`.
```

> To enable pre-warmed app start tracking:

Reviewer: ```suggestion
To enable prewarmed app start tracking:
```

Cold and warm start are Mobile Vitals, which you can learn about in the [full documentation](/product/performance/mobile-vitals).

### Pre-warmed app start tracking

> Starting with iOS 15, the system might [pre-warm](https://developer.apple.com/documentation/uikit/app_and_environment/responding_to_the_launch_of_your_app/about_the_app_launch_sequence#3894431) your app by creating the process before the user opens it.
> You can enable the experimental feature `enablePreWarmedAppStartTracking` to report pre-warmed app starts. Once enabled, the SDK will drop the first app start spans if pre-warming paused during these steps. This approach will shorten the app start duration, but it represents the duration a user has to wait after clicking the app icon until the app is responsive.

Reviewer 2: ```suggestion
Starting with iOS 15, the system might [prewarm](https://developer.apple.com/documentation/uikit/app_and_environment/responding_to_the_launch_of_your_app/about_the_app_launch_sequence#3894431) your app by creating the process before the user opens it. In such cases, we won’t be able to reliably measure the app start. But with [sentry-cocoa 7.30.3](https://github.com/getsentry/sentry-cocoa/releases/tag/7.30.3), we’ve introduced a `enablePreWarmedAppStartTracking` feature, (still in its experimental phase), which allows us to collect prewarmed app starts.

Once enabled, the SDK will drop the first app start span if prewarming pauses. This approach will shorten the app start duration, but it will accurately represent the span of time from when a user clicks the app icon to when the app is responsive. 
```
I'm not sure I'm understanding the above accurately. What I think it means is that the `enablePreWarmedAppStartTracking` feature will make the total application launch time appear shorter than it actually is (because it will drop the first app start span). But it will accurately report the rest of the amount of time it takes for the app to launch (maybe from runtime to UIWindowDidBecomeVisibleNotification?) Let me know if this is not accurate.

Author: Your interpretation is correct @lizokm 😀

> With this feature, the SDK differentiates between four different app start types:
> 
> * __Cold start__: Same as _cold start_ above, and __not__ prewarmed.
> * __Warm start__: Same as _warm start_ above, and __not__ prewarmed.
> * __Cold start pre-warmed__: Same as _warm start_, and prewarmed.
> * __Warm start pre-warmed__: Same as _warm start_, and prewarmed.

Reviewer 2: ```suggestion
With this feature, the SDK differentiates between four different app start types:

* __Non-prewarmed cold start__ (See _cold start_  definition above.)
* __Non-prewarmed warm start__ (See _warm start_  definition above.)
* __Prewarmed cold start__
* __Prewrmed warm start__
```
